src/goose/brokers/redis/batch.clj

```diff
@@ -70,3 +58,72 @@
        :retrying    retrying
        :successful  successful
        :dead        dead})))
+
+(defn set-batch-expiration
+  [conn id linger-in-hours]
+  (let [batch-state-key (d/prefix-batch id)
+        enqueued-job-set (d/construct-batch-job-set id d/enqueued-job-set)
+        retrying-job-set (d/construct-batch-job-set id d/retrying-job-set)
+        successful-job-set (d/construct-batch-job-set id d/successful-job-set)
+        dead-job-set (d/construct-batch-job-set id d/dead-job-set)
+        linger-in-sec (u/hour->sec linger-in-hours)]
+    (redis-cmds/expire conn batch-state-key linger-in-sec)
+    (redis-cmds/expire conn enqueued-job-set linger-in-sec)
+    (redis-cmds/expire conn retrying-job-set linger-in-sec)
+    (redis-cmds/expire conn successful-job-set linger-in-sec)
+    (redis-cmds/expire conn dead-job-set linger-in-sec)))
+
+(defn post-batch-exec
+  [conn id]
+  (car-locks/with-lock conn "post-batch-exec"
+                       d/redis-batch-lock-timeout-ms
+                       d/redis-batch-lock-wait-ms
+                       (let [{:keys [batch-state] :as batch} (get-batch-state conn id)
+                             status (batch/status-from-counts batch)
+                             {:keys [callback-job-id
+                                     callback-fn-sym
+                                     linger-in-hours
+                                     queue
+                                     ready-queue
+                                     retry-opts]} batch-state]
+                         (when (= status batch/status-complete)
+                           (when (and
+                                   (some? callback-fn-sym)
+                                   (not callback-job-id))
+                             (let [callback-args (-> (select-keys batch [:successful :dead])
+                                                     (list)
+                                                     (conj id))
+                                   callback-job (-> (job/new callback-fn-sym
+                                                             callback-args
+                                                             queue
+                                                             ready-queue
+                                                             retry-opts)
+                                                    (assoc :batch-id id))]
+                               (redis-cmds/enqueue-back conn ready-queue callback-job)
+                               (redis-cmds/hset conn (d/prefix-batch id) (name :callback-job-id) (:id callback-job))))
+                           (let [linger-in-hours (if linger-in-hours
```

Comment:
Can move this if condition logic to the multi-parameter `set-batch-expiration` above.

Reply:
+1 to the suggestion, can also use if-let here

Reply:
Moved it to `set-batch-expiration`. However using if-let might not be feasible, since the rest of the code is common to both cases.